### DISCUSSION TITLE
How to wrap useToast in a seperate class (not inside setup)

### DISCUSSION THREAD

**Discussion by boindil** [cdmoro/bootstrap-vue-3#1829] [Q&A] [answered]
**What you are trying to do.** I would like to use a seperate class, lets call it ToastHelper that handles all my toasts.

**Any error messages you are getting.** `[Vue warn]: inject() can only be used inside setup() or functional components.`

**Environment (please complete the following information):**
- Package Version: 0.16.6

Is there any way to achieve that goal?

**Comment by sceee** [comment, marked as answer]
Hi @boindil ,

As `useToast` is a composable, it ["should only be called in `<script setup>` or the `setup()` hook."](https://vuejs.org/guide/reusability/composables#usage-restrictions)

But you can nest composables to probably achieve what you want (reuse the same configuration across multiple components).

Say you create a `ToastHelper.ts` with the following content:
```typescript
import { useToast } from 'bootstrap-vue-next';

export function useSuccessToast() {
  const toast = useToast();

  const showSuccessToast = () =>
    toast.show?.({ props: { title: 'Hello', body: 'World' } });

  return { show: showSuccessToast };
}
```

And use your own `useSuccessToast` composable in your components:
```vue
<template>
  <BButton @click="onClickDiv">
    {{ msg }}
  </BButton>
</template>
<script setup lang="ts">
import { ref } from 'vue';
import { useSuccessToast } from './TestToast';

const toast = useSuccessToast();

const msg = ref('Comp');

const onClickDiv = () => {
  toast.show();
};
</script>

```

**Comment by boindil** [comment]
Hi @sceee,

thranks, thats working great! What I was wondering is whether its not possible to output html (or elements using `h`) in the body part this way.

I tried using a simple `<p>test</p>` as body and its not working sadly.

**Reply by VividLemon** [reply]
> thranks, thats working great! What I was wondering is whether its not possible to output html (or elements using h) in the body part this way.

https://bootstrap-vue-next.github.io/bootstrap-vue-next/docs/composables/useToast.html#advanced-usage
